File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque
def solution(s):
    answer = True
    q = deque()
    for i in s:
        if i == "(":
            q.append(i)
        else:
            if len(q) == 0:
                return False
            q.pop()

    return True if len(q) == 0 else False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <stack>

using namespace std;

bool solution(string s)
{
    stack<char> my_stack;

    for (int i = 0; i < s.size(); i++) {
        char c = s[i];
        if (c == '(') {
            my_stack.push(c);
        } else {
            if (my_stack.empty()) {
                return false;
            }
            my_stack.pop();
        }
    }

    return my_stack.empty();
}